conda_build/utils.py

```diff
@@ -1007,9 +1007,20 @@ def get_stdlib_dir(prefix, py_ver):
         lib_dir = os.path.join(prefix, 'Lib')
     else:
         lib_dir = os.path.join(prefix, 'lib')
-        python_folder = glob(os.path.join(lib_dir, 'python?.*'))
-        if python_folder:
-            lib_dir = os.path.join(lib_dir, python_folder[0])
+        python_folders = glob(os.path.join(lib_dir, 'python?.*'))
+        if python_folders:
+            # There might be a symlink python3.1 -> python3.10
+            # In that case, select last entry to ensure we find python3.10
+            # References:
+            # - https://github.com/conda/conda-build/issues/4477
+            # - https://github.com/conda/conda/issues/10969
+            # - https://github.com/conda/conda/issues/11423
+            # - https://github.com/conda/conda/pull/10478
+            # - https://github.com/conda/conda/pull/10970
+            # - https://github.com/conda-forge/python-feedstock/pull/511
+            # - https://github.com/AnacondaRecipes/python-feedstock/pull/47
+            python_folder = sorted(python_folders)[-1]
+            lib_dir = os.path.join(lib_dir, python_folder)
```

Comment:
We feel uncomfortable solely relying on a sort here to identify the correct lib folder. This feels potentially fragile in the future. Let's filter out symlinks and grab the first as was done previously:

```python
python_folder = next(
    (
        path
        for path in glob(os.path.join(lib_dir, "python?.*"))
        if not os.path.islink(path)
    ),
    None,
)
if python_folder:
    lib_dir = os.path.join(lib_dir, python_folder)
```